Update README.md and NEWS.md

dongahn opened the pull request:
This PR updates README.md and NEWS.md in preparation for v0.8.0 tag and release

Address 2 issues:
- Issue #545 (@andre-merzky: So if you can give me a feedback on the new README.md, this should be very useful);
- Issue #544 (@cmisale: once this gets in, we should be able to tag/release 0.8.0 so that you can use a tagged name for your Kube+Flux work);

dongahn commented:
@milroy: as we discussed at our meeting with IBM, in case I don't get to complete all of the release checklist, this page has the remaining work: https://github.com/flux-framework/flux-core/wiki/Release-Checklist

andre-merzky commented on README.md:
````diff
@@ -93,39 +106,36 @@ flag to each flux command below.
-export LUA_CPATH="$HOME/flux-sched/rdl/?.so;${LUA_CPATH};;"
-export FLUX_MODULE_PATH=$HOME/flux-sched/sched
 $HOME/local/bin/flux start -s3
 ```
````
Not sure if this is the right place, but I got initially stumped by the fact that `flux start` opens a new shell session - it might be useful to mention that to avoid surprises.

dongahn replied:
Absolutely. Good feedback, I will update this at the airport.

andre-merzky commented:
@dongahn : thanks for the ping, that looks good to me.  FWIW, I found `flux env` to be very useful to be able to connect to the same flux instance from a different shell.  Once I discovered that (and specifically the `FLUX_URI`, things became much clearer.  Not sure if the README is the right place for that info though...

dongahn commented:
@andre-merzky: impressive how much progress you have made in a short amount of time. 

Yeah, `flux env` and `flux proxy` would be good targets for our workflow examples. @cmoussa1 may be interested in working on a new example based on your input...

dongahn commented:
Ok I made a requested change. And I took liberty of forcing a push so that this PR can go in with no further squashing  if @milroy or @SteVwonder approves it.

dongahn commented:
Ok. I rebased this to the latest master that has both PR #548 and #543  and updated NEWD.md. I forced a push so this doesn't need to be squashed further.

milroy commented on NEWS.md:
Typo: resource

milroy commented on README.md:
```diff
@@ -1,26 +1,41 @@
+The `resource` module uses a graph to represent resources of arbitrary types
+as well as their complex relationships and to match the highly sophisticated
+resource requirements of a Flux jobspec to the compute and other resources
+on this graph. Both of these modeuls are loaded into a Flux instance and
```
Typo: modules

milroy commented on README.md:
```diff
@@ -1,26 +1,41 @@
+
+flux-sched introduces queuing and resource matching services to extend Flux
+to provide advanced batch scheduling. Jobs are submitted to flux-sched via
+`flux job submit` which then added to our queues for scheduling.
```
typo: which are then added

dongahn commented:
Ouch. Thanks Dan. I will get to this as soon as I get my computer back.

dongahn commented:
@milroy: OK. I addressed all of the issues you identified and forced a push. This should be good to go in once Travis CI turns green. Thanks!

dongahn commented:
I took the liberty to merge in @chu11's PR #549 and rebased this PR to that. I added one more item to `NEWS.md` and forced a push once again. This should be good to go in.

dongahn commented:
Thanks!
